src/api-resource.js

```diff
@@ -0,0 +1,60 @@
+import axios from "axios";
+import { apiUrl } from "./constants.js";
+
+/**
+ * Create a resource object to interact with Userfront API.
+ * @param {Object} options
+ * @property {String} options.store Userfront.store object containing `tenantId` & `accessToken`
+ * @property {String} [options.basePath] Base path of resource
+ * @property {String} options.path Sub-path of resource
+ * @property {String|Number} options.id Identifier of resource
+ * @property {Function} options.afterCreate Function to call after create()
+ * @property {Function} options.afterRead Function to call after read()
+ * @property {Function} options.afterUpdate Function to call after update()
+ * @property {Function} options.afterDelete Function to call after delete()
+ */
+export default (options = {}) => {
+  const { store, path, id } = options;
+  const { tenantId, accessToken } = store;
+  if (!tenantId) {
+    throw new Error("API resource error: Missing tenant ID");
+  }
+  if (!accessToken) {
+    throw new Error("API resource error: Missing access token");
+  }
+  if (!path) {
+    throw new Error("API resource error: Missing path");
+  }
+
+  let { basePath } = options;
+  if (!basePath) {
+    basePath = `${apiUrl}tenants/${tenantId}`;
+  }
```

Comment:
Currently `tenantId` and `accessToken` are required. If we decide to go forward with the `apiResource` design, I'd probably make another factory function e.g. `apiBase` (not sure of the name) that `apiResource` could adopt functionality from here and remove the need to pass those as options.